cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto

```diff
@@ -157,6 +159,35 @@ message CreateInstalledPackageRequest {
   ReconciliationOptions reconciliation_options = 6;
 }
 
+// UpdateInstalledPackageRequest
+//
+// Request for UpdateInstalledPackage. Partial resource updates are supported. 
+// For example, to change the package version one only needs to specify the version reference. 
+// Similarly to update the values, one only needs to specify that field
+message UpdateInstalledPackageRequest {
+  // A reference uniquely identifying the installed package being updated.
+  // Required
+  InstalledPackageReference installed_package_ref = 1;
+
+  // The target context where the package is intended to be installed. Optional
+  Context target_context = 2;
```

Comment:
This `target_context` shouldn't be required should it? I mean, the `installed_package_ref` gives us the context where the package is already installed, we're just updating it (in the same place).

Reply:
On another note, something I did expect to see in this request is an `AvailablePackageReference` with the uniq ref of the chart that will be used in the upgrade?

Reply:
Its not required. It says "Optional" in the description. line 172

Reply:
> On another note, something I did expect to see in this request is an `AvailablePackageReference` with the uniq ref of the chart that will be used in the upgrade?

Are you sure about this? I mean after all this is an "update". By letting them specify the chart that may change the whole semantic of the installed package, e.g. 
1. create mongodb
2. update (1) to redis

Does that make sense?

Reply:
yeah, I think I'd rather start without it. Need to think about it some more and see what flux does. We can always add it later

Reply:
> > Its not required. It says "Optional" in the description. line 172
> 
> Right, sorry, I shouldn't have used the word "required". I meant, why is it even here, or how could it even be used? I mean, if my installed application is in the namespace "michaels-ns', and I specify in the upgrade that the target context is the same cluster but "gregs-ns", what do you expect to happen?
> 
> Remember target_context is where flux will place the artifacts for the HelmRelease (deployments, pods, secrets, etc). I don't see any documentation on flux site saying I cannot update this field. Although I have not tried it, I would expect them to delete what they created in one namespace and re-create it in another? Should I give it a try before confirming?

I tried, it works, not 100% the way I expected:
```
$ flux create source helm podinfo \
>     --url https://stefanprodan.github.io/podinfo \
>     --namespace default
✚ generating HelmRepository source
► applying HelmRepository source
✔ source created
◎ waiting for HelmRepository source reconciliation
✔ HelmRepository source reconciliation completed
✔ fetched revision: 8411f23d07d3701f0e96e7d9e503b7936d7e1d56
[v1alpha1]$ flux create hr my-podinfo \
>   --namespace=kubeapps \
>   --target-namespace=test \
>   --create-target-namespace=true \
>   --source=HelmRepository/podinfo.default \
>   --chart=podinfo
✚ generating HelmRelease
► applying HelmRelease
✔ HelmRelease created
◎ waiting for HelmRelease reconciliation
✔ HelmRelease my-podinfo is ready
✔ applied revision 6.0.0
[v1alpha1]$ kubectl get all -n test
NAME                                   READY   STATUS    RESTARTS   AGE
pod/test-my-podinfo-6589894fb4-5s8gl   1/1     Running   0          112s

NAME                      TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)             AGE
service/test-my-podinfo   ClusterIP   10.96.42.177   <none>        9898/TCP,9999/TCP   112s

NAME                              READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/test-my-podinfo   1/1     1            1           112s

NAME                                         DESIRED   CURRENT   READY   AGE
replicaset.apps/test-my-podinfo-6589894fb4   1         1         1       112s
[v1alpha1]$ kubectl edit helmrelease/my-podinfo -n kubeapps
helmrelease.helm.toolkit.fluxcd.io/my-podinfo edited
[v1alpha1]$ kubectl get ns
NAME                 STATUS   AGE
default              Active   26h
flux-system          Active   26h
ingress-nginx        Active   26h
kube-node-lease      Active   26h
kube-public          Active   26h
kube-system          Active   26h
kubeapps             Active   26h
local-path-storage   Active   26h
test                 Active   2m53s
test-2               Active   7s
[v1alpha1]$ kubectl get ns
NAME                 STATUS   AGE
default              Active   26h
flux-system          Active   26h
ingress-nginx        Active   26h
kube-node-lease      Active   26h
kube-public          Active   26h
kube-system          Active   26h
kubeapps             Active   26h
local-path-storage   Active   26h
test                 Active   2m56s
test-2               Active   10s
[v1alpha1]$ kubectl get all -n test-2
NAME                                     READY   STATUS    RESTARTS   AGE
pod/test-2-my-podinfo-56d4c876d9-wv5df   1/1     Running   0          20s

NAME                        TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)             AGE
service/test-2-my-podinfo   ClusterIP   10.96.101.106   <none>        9898/TCP,9999/TCP   20s

NAME                                READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/test-2-my-podinfo   1/1     1            1           20s

NAME                                           DESIRED   CURRENT   READY   AGE
replicaset.apps/test-2-my-podinfo-56d4c876d9   1         1         1       20s
[v1alpha1]$ kubectl get all -n test
NAME                                   READY   STATUS    RESTARTS   AGE
pod/test-my-podinfo-6589894fb4-5s8gl   1/1     Running   0          3m12s

NAME                      TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)             AGE
service/test-my-podinfo   ClusterIP   10.96.42.177   <none>        9898/TCP,9999/TCP   3m12s

NAME                              READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/test-my-podinfo   1/1     1            1           3m12s

NAME                                         DESIRED   CURRENT   READY   AGE
replicaset.apps/test-my-podinfo-6589894fb4   1         1         1       3m12s
```
in other words, flux will put the artifacts into a new namespace but won't clean up the old ones...

Reply:
It sounds like it's just installing the helm chart in a new namespace and leaving the old one orphaned. Certainly not possible with Helm itself - you can only upgrade a release that already exists (in the namespace). Up to you. My pref would be to not include it (we'll explicitly not use it in the helm direct plugin, and it seems that even with flux, it's not doing an upgrade operation it seems). We can always add it later if we find there's a need for it.

Reply:
ok, I don't have a strong opinion. I'll remove it and push. I assume that was the last remaining item. So after all successful checks I will merge

Reply:
Thanks. Yeah, the `helm` CLI uses the term `upgrade` in the cli, but it's confusing as it's used both to upgrade a chart version or update the values/config of a release without changing the version. That's why we went with "update" in our API rather than upgrade.

Checks have passed, I'll merge so I can pull it into my branch. Thanks!

Reply:
sounds good. Thanks